lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -41,7 +41,7 @@ function wp_theme_has_theme_json( $clear_cache = false ) {
 	 * Clean theme.json related cached data.
 	 */
 	function wp_theme_clean_theme_json_cached_data() {
```

Comment:
`wp_theme_clean_theme_json_cached_data` is now called in the following filters: `switch_theme`, `start_previewing_theme`. Do we need any other?

What if a theme is updated and removes or adds a `theme.json`? How can we flush the cache at that point?

Reply:
Cleaning the cache on [delete_theme](https://developer.wordpress.org/reference/hooks/delete_theme/) sounds like a good idea.

Reply:
@oandregal +1, we should definitely delete the cache when the theme is updated.

If you're referring to adding/modifying `theme.json` during development of a theme for example, I think for that we can't properly handle cache, so we may want to wrap the cache usage here in a condition to only use cache if `! WP_DEBUG`.

Reply:
> to only use cache if ! WP_DEBUG

Nice, I'm going to implement this.

I've seen something similar in `wp_get_global_stylesheet` ([see](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/global-styles-and-settings.php#L88)), though it also includes `SCRIPT_DEBUG` and other stuff. I'm not really sure why it was added. It sounds more sensible to just use `WP_DEBUG`.

Reply:
Cleaned the cache upon upgrading the theme at https://github.com/WordPress/gutenberg/pull/45543/commits/13510c70e306f0d5f708e792d3ec7a6717200a30

Did not added the same for the `deleted_theme` action because users cannot delete the active theme (manually or via the wp-cli).